Common/Securities/SecurityExchangeHours.cs

```diff
@@ -271,6 +296,57 @@ public DateTime GetNextMarketOpen(DateTime localDateTime, bool extendedMarket)
             throw new ArgumentException("Unable to locate next market open within two weeks.");
         }
 
+        /// <summary>
+        /// Checks whether the current marketOpen time is part of the last market open segment
+        /// </summary>
+        /// <param name="marketHours">Trading segments of the current time</param>
+        /// <param name="time">The current time</param>
+        /// <param name="marketOpen">Current next market open time</param>
+        /// <param name="extendedMarket">Flag to indicate whether PreMarket time is considered as open market or not</param>
+        /// <returns>True if the current next market open time is part of the last market open segment. False otherwise.</returns>
+        public bool IsContinuousOpenMarket(LocalMarketHours marketHours, DateTime time, DateTime marketOpen, bool extendedMarket)
+        {
+            var index = marketHours.Segments.ToList().FindIndex(x => x.Start == marketOpen.TimeOfDay);
+            DateTime lastMarketOpen = new DateTime();
+            TimeSpan? lastSegment;
+            if (index != 0)
+            {
+                // There's still one segment before the current one in the day
+                lastSegment = marketHours.Segments.ElementAt(index - 1).End;
+
+                if (marketHours.Segments.ElementAt(index - 1).State == MarketHoursState.Market || 
+                    (marketHours.Segments.ElementAt(index - 1).State == MarketHoursState.PreMarket && extendedMarket))
+                {
+                    lastMarketOpen = time.Date + lastSegment.Value;
+                }
+            }
+            else
+            {
+                // There's no segment before the current one in the day, then take the yesterday's last segment
+                var oneDayBefore = time.Date.AddDays(-1);
+                var oneDayBeforeMarketHours = GetMarketHours(oneDayBefore.DayOfWeek);
+                lastSegment = oneDayBeforeMarketHours.Segments.Any() ?
+                    oneDayBeforeMarketHours.Segments.Last().End : null;
+
+                if (lastSegment.HasValue)
+                {
+                    if (oneDayBeforeMarketHours.Segments.Last().State == MarketHoursState.Market ||
+                    (marketHours.Segments.Last().State == MarketHoursState.PreMarket && extendedMarket))
+                    {
+                        lastMarketOpen = time.Date.AddDays(-1) + lastSegment.Value;
+                    }
+                }
+            }
+
+            // Check the current market open time is not part or continuation of the last market open segment
+            if (marketOpen - lastMarketOpen != new TimeSpan())
+            {
+                return false;
+            }
+
+            return true;
+        }
+
```

Comment:
Cool! Seems that the changes here could be handled by LocalMarketHours internally instead, maybe a missing functionality/method , I think it would allow for a cleaner usage of the class and potentially simplify the implementation it self too, dropping all linq usage and some duplication